Lib/_strptime.py

```diff
@@ -250,12 +250,30 @@ def pattern(self, format):
         format = regex_chars.sub(r"\\\1", format)
         whitespace_replacement = re_compile(r'\s+')
         format = whitespace_replacement.sub(r'\\s+', format)
+        year_in_format = False
+        day_of_month_in_format = False
         while '%' in format:
             directive_index = format.index('%')+1
+            format_char = format[directive_index]
             processed_format = "%s%s%s" % (processed_format,
                                            format[:directive_index-1],
-                                           self[format[directive_index]])
+                                           self[format_char])
             format = format[directive_index+1:]
+            match format_char:
+                case 'Y' | 'y' | 'G':
+                    year_in_format = True
+                case 'd':
+                    day_of_month_in_format = True
+        if day_of_month_in_format and not year_in_format:
+            import warnings
+            warnings.warn("""\
+Parsing dates involving a day of month without a year specified is ambiguious
+and fails to parse leap day. The default behavior will change in Python 3.15
+to either always raise an exception or to use a different default year (TBD).
+To avoid trouble, add a specific year to the input & format.
+See https://github.com/python/cpython/issues/70647.""",
+                          DeprecationWarning,
+                          stacklevel=5)
```

Comment:
The `5` looks fragile, but I don't see a good way around it, short of teaching `warnings` new tricks.
Hopefully this code won't get refactored in the next few years.

Reply:
Oh, wow, [nice job with the time machine](https://github.com/python/cpython/pull/100840) :)
```suggestion
                          skip_file_prefixes=(os.path.dirname(__file__),))
```
Looks like this would be the first use of `skip_file_prefixes` in the stdlib.